### ISSUE TITLE
Support more fragment transitions

### ISSUE THREAD

**Issue by toolforger** [asciidoctor/asciidoctor-reveal.js#378]
https://asciidoctor.org/docs/asciidoctor-revealjs/#fragments has this to say:

> Note that only `fade-in` is supported for lists at the moment.

Can I do anything for getting support all fragment transitions that reveal.js offers?
(I am a software developer but unfortunately not in Ruby; is there anything else I can do?)

**Comment by Mogztter** [comment]
Hi @toolforger 

I think the documentation is inaccurate or maybe I'm missing something.
I don't see why only `fade-in` would be supported. reveal.js is using classes to define transitions: https://revealjs.com/fragments/

The following should work:

```adoc
= Presentation

== First Slide

[.fragment]
Fade in

[.fragment.fade-out]
Fade out

[.fragment.highlight-red]
Highlight red

[.fragment.fade-in-then-out]
Fade in, then out

[.fragment.fade-up]
Slide up while fading in
```

Could you please confirm that this is working? If so, and if you are still willing to help, you can update the documentation :smile: 

Thanks!

**Comment by Mogztter** [comment]
> I don't see why only fade-in would be supported. reveal.js is using classes to define transitions: revealjs.com/fragments

Now I understand why only `fade-in` is supported for lists because we use the fragment class on list items (not on the list block) and we don't configure which transition we want. So by default, `fade-in` is used.

For reference, roles are applied on the list block (not on list items). As far as I know, AsciiDoc does not allow to declare roles on list items.

A first step, would be to configure the transition globally (i.e., for all the list items). What do you think?

**Comment by mojavelinux** [comment]
> As far as I know, AsciiDoc does not allow to declare roles on list items.

That's correct. It does not. (It's still in the proposal phase) However, what I've done in other converters as a workaround is to apply the role around the text and then promote that to the list item itself.

**Comment by Mogztter** [comment]
> However, what I've done in other converters as a workaround is to apply the role around the text and then promote that to the list item itself.

Just to be sure, that would be:

```
== Topic

[%step]
* [.highlight-blue]#Blue#
* White
* [.highlight-red]#Red#
```

**Comment by ggrossetie** [comment]
Revisiting this — the docs have since been substantially rewritten (most recently in #553) and are accurate now. The [Steps doc page](https://docs.asciidoctor.org/reveal.js-converter/latest/converter/syntax/fragment/) confirms what @ggrossetie found in this thread: block and inline fragments already support every reveal.js transition style (`fade-out`, `fade-up`, `highlight-red`, `grow`, etc. — full table on that page), it's only the list-*item* case (`[%step]` on a `* list`) that's limited to `fade-in`, since AsciiDoc doesn't support setting a role per list item. That's now called out explicitly as a documented limitation, with the `[.step.highlight-red]#...#` inline-role workaround (from @mojavelinux's suggestion above) and the "reveal all at once" open-block workaround both included on the same page.

Closing as resolved by the doc rewrite — the per-list-item role promotion mojavelinux sketched out is still a real, un-implemented enhancement if someone wants to pick it up, but it'd be its own scoped feature request rather than a continuation of this one.
